### ISSUE TITLE
[AlphaHoverCard] Content height increase in dynamic hovercard causes layout shift 

### ISSUE THREAD

**Issue by chloerice** [Shopify/polaris-react-archive#11614]
## Tl;dr: 
When height of overlay contents increases beyond available height, transition from initial render while measuring to next measured and correct height and position causes overflow if positioned below toward bottom of visible page and temporary scrollbar appears that causes miscalculation of horizontal position equivalent to width of scrollbar.

## Problem

The hovercard determines where to position itself and in what direction based on available space. It renders initially with opacity 0 while it's `measuring` (based on `overlayDetails` passed into the PositionedOverlay.render callback provided). If there is not scrollbar on page and the height increases when horizontally (left or right) positioned and calculated vertical position is below, if the activator is low enough on the page the overlay will render causing overflow momentarily, and horizontally mis-position itself into the activator by the width of the scrollbar.

### Expected behavior

- Calculates the height based on available space while not visible
- Does not overflow the page when initially visible
- Correctly positions itself horizontally (with `right` if positioned left of activator, `left` if positioned right of activator)

https://github.com/Shopify/polaris/assets/18447883/6c92291f-87b6-4772-830e-89dc20df17a1

### Actual behavior

- Pushes the page into overflow-y scroll momentarily
- Positions itself with scrollbar width added to the left or right position value

https://github.com/Shopify/polaris/assets/18447883/d57a513e-9d98-470a-8bb4-d635be46c645

### Steps to reproduce

See latest Storybook publish of the [AlphaHoverCard PR](https://github.com/Shopify/polaris/pull/10504) or 
locally run Storybook and adjust the map on line 577 of AlphaHoverCard.stories.tsx to spread several (like [...orders, ...orders, ...orders, ...orders]) so that the table has enough rows to reproduce.
